crates/bevy_ecs/src/query/state.rs

```diff
@@ -41,16 +41,30 @@ impl<Q: WorldQuery, F: WorldQuery> FromWorld for QueryState<Q, F> {
 }
 
 impl<Q: WorldQuery, F: WorldQuery> QueryState<Q, F> {
-    pub(crate) fn as_readonly(&self) -> &QueryState<Q::ReadOnly, F::ReadOnly> {
+    /// Converts this `QueryState` reference to a `QueryState` that does not access anything mutably.
+    pub fn as_readonly(&self) -> &QueryState<Q::ReadOnly, F::ReadOnly> {
```

Comment:
Oh excellent. Can you highlight this in the change log? This will be genuinely useful to end users who are trying to define helper functions that consume queries.

Reply:
Although admittedly this would be much more useful as a method on `Query` itself 🤔

Reply:
Yeah this is just for `QueryState` so really this isnt gonna be useful for anyone not doing direct world access. I'd love to actually provide an api for `Query::as_readonly` though. but i dont wanna do it in this PR. I think putting it in the changelog would probably give people the wrong idea and get excited thinking `Query` could do it :rofl:

Reply:
Yep, sounds good to me.

Reply:
for extra info: while implementing this would be pretty simple, the lifetimes involved are a bit subtle so we'd want some compile fail tests and proper scrutiny of the added methods. If added in this PR the compile-fail tests would take up a huge % of the diff :rofl: and it likely wouldnt get the desired review since its lumped in with a bunch of other stuff.